src/geometry/manifold/real_instances.lean

```diff
@@ -368,8 +368,17 @@ begin
   constructor
 end
 
-/- Register an instance of `0 < 1` to be able to use freely the manifold structure on `Icc 0 1`,
-and also its zero and one. -/
-instance : fact ((0 : ℝ) < 1) := zero_lt_one
+/- Register the manifold structure on `Icc 0 1`, and also its zero and one. -/
+section
+
+lemma fact_zero_lt_one : fact ((0 : ℝ) < 1) := zero_lt_one
```

Comment:
Should this be private?

Reply:
You never know if someone might need it later, so not adding private gives us more freedom, at essentially no cost.

Reply:
OK, fair enough. I think I was confused whether this could pop up unwanted in `library_search`, but in hindsight it's pretty unlikely someone would be using it to search for `fact ...`.